frontend/packages/modelServing/extension-points/index.ts

```diff
@@ -11,6 +33,14 @@ export type ModelServingPlatformExtension = Extension<
       enabledLabel: string;
       enabledLabelValue: string;
     };
+    deployments: {
+      watch: CodeRef<
+        (
+          project: ProjectKind,
+          opts?: K8sAPIOptions,
+        ) => [Deployment[] | undefined, boolean, Error | undefined]
```

Comment:
Why not make the result always an array?
```suggestion
        ) => [Deployment[], boolean, Error | undefined]
```